### PR TITLE
Improve hat graph example

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Extract a `hat_graph()` function, that can be readily copy/pasted by end-users.

It would also be good to have some other data semantics. Hat graphs are designed to compare datasets with respect to a reference dataset. For players A/B, it seems a bit arbitrary that Player A is considered the reference player. Unfortunately, I currently cannot think of a good domain/data semantics. Suggestions welcome.

### CODE REVIEW

File: examples/lines_bars_and_markers/hat_graph.py [hunk 1]
```diff
@@ -10,42 +10,65 @@
 import matplotlib
 import numpy as np
 import matplotlib.pyplot as plt
+
```
Comment (jklymak) [inline examples/lines_bars_and_markers/hat_graph.py:13]: Can you fix the comment above while we are here?  Refer ... to: "For discussion of hat graphs, see https://doi.org/10.1186/s41235-019-0182-3"

**Review by jklymak — APPROVED** [review]
This looks good.  Might try and fix the opening couple of sentences to have better grammar...

File: examples/lines_bars_and_markers/hat_graph.py (hunk 1 above)
Comment (jklymak) [reply examples/lines_bars_and_markers/hat_graph.py:13]: Please feel free to self merge either way

**Review by QuLogic — COMMENTED** [review]
I seem to have forgotten to post this.

File: examples/lines_bars_and_markers/hat_graph.py (hunk 1 above)
Comment (timhoffm) [reply examples/lines_bars_and_markers/hat_graph.py:13]: I've inlined the link into the first sentence by linking "hat graph" there.

**Review by QuLogic — APPROVED** [review]
Can merge if CI passes.

**Pull request merged**